core/iwasm/libraries/libc-wasi/sandboxed-system-primitives/src/posix.c

```diff
@@ -322,36 +363,149 @@ wasi_addr_ip_to_bh_ip_addr_buffer(__wasi_addr_ip_t *addr,
 __wasi_errno_t
 wasmtime_ssp_clock_res_get(__wasi_clockid_t clock_id,
                            __wasi_timestamp_t *resolution)
-{
+
 #ifdef BH_PLATFORM_WINDOWS
```

Comment:
I see the implementation for posix and windows platforms is quite different. I'd suggest we:

1. add new functions to the core/shared/platform/common/posix/posix_time.c: `os_clock_res_get` and `os_clock_time_get()` and move the original implementation from this file to those functions
2. `os_clock_res_get` and `os_clock_time_get()`  to core/shared/platform/windows/win_time.c and move your windows-specific implementation there
3. in here, call the os_clock_res_get and os_clock_time_get functions instead